crates/miden-lib/src/account/faucets/network_fungible.rs

```diff
@@ -0,0 +1,294 @@
+            Felt::from(network_faucet.faucet.decimals()),
+            network_faucet.faucet.symbol().into(),
+            Felt::ZERO,
+        ]);
+
+        // Convert AccountId to Word representation for storage
+        let owner_account_id_word: Word = [
+            Felt::new(0),
+            Felt::new(0),
+            network_faucet.owner_account_id.suffix(),
+            network_faucet.owner_account_id.prefix().as_felt(),
+        ]
+        .into();
+
+        // Second storage slot stores the owner account ID
+        let owner_slot = StorageSlot::Value(owner_account_id_word);
+
+        AccountComponent::new(
+            network_fungible_faucet_library(),
+            vec![StorageSlot::Value(metadata), owner_slot]
+        )
+            .expect("network fungible faucet component should satisfy the requirements of a valid account component")
+            .with_supported_type(AccountType::FungibleFaucet)
+    }
+}
+
+impl TryFrom<Account> for NetworkFungibleFaucet {
+    type Error = FungibleFaucetError;
+
+    fn try_from(account: Account) -> Result<Self, Self::Error> {
+        let account_interface = AccountInterface::from(&account);
+
+        NetworkFungibleFaucet::try_from_interface(account_interface, account.storage())
+    }
+}
+
+impl TryFrom<&Account> for NetworkFungibleFaucet {
+    type Error = FungibleFaucetError;
+
+    fn try_from(account: &Account) -> Result<Self, Self::Error> {
+        let account_interface = AccountInterface::from(account);
+
+        NetworkFungibleFaucet::try_from_interface(account_interface, account.storage())
+    }
+}
+
+/// Creates a new faucet account with network fungible faucet interface,
+/// account storage type, specified authentication scheme, and provided meta data (token symbol,
+/// decimals, max supply, owner account ID).
+///
+/// The network faucet interface exposes two procedures:
+/// - `distribute`, which mints an assets and create a note for the provided recipient.
+/// - `burn`, which burns the provided asset.
+///
+/// Both `distribute` and `burn` can only be called from note scripts. `distribute` requires
+/// authentication. The authentication procedure is defined by the specified authentication scheme.
+/// `burn` does not require authentication and can be called by anyone.
+///
+/// The storage layout of the network faucet account is:
+/// - Slot 0: Reserved slot for faucets.
+/// - Slot 1: Public Key of the authentication component.
+/// - Slot 2: [num_tracked_procs, allow_unauthorized_output_notes, allow_unauthorized_input_notes,
+///   0].
+/// - Slot 3: A map with tracked procedure roots.
+/// - Slot 4: Token metadata of the faucet.
+/// - Slot 5: Owner account ID.
+pub fn create_network_fungible_faucet(
+    init_seed: [u8; 32],
+    symbol: TokenSymbol,
+    decimals: u8,
+    max_supply: Felt,
+    owner_account_id: AccountId,
+    account_storage_mode: AccountStorageMode,
+    auth_scheme: AuthScheme,
+) -> Result<Account, FungibleFaucetError> {
```

Comment:
A couple of questions:

- Why do we need to pass in `account_storage_mode` here? Shouldn't `AccountStorageMode::Network` be the only possible option here?
- Similarly, do we need to pass in the `auth_scheme`? Seems like the only allowed scheme is `NoAuth` - and so maybe we could just default to that?